src/extension_utils/utils.ts

```diff
@@ -258,37 +258,35 @@ export const checkConfig = (configName: string): boolean => {
 
 export const checkPythonDependencies = async (context: vscode.ExtensionContext, pythonExecutable: string) => {
   let hasInstalledDependencies: boolean = false;
+  const pathToLibs: string = getPathToScript(context, "out", "python_libs");
   if (checkPipDependency() && checkPythonDependency()) {
     if (checkConfig(CONFIG.SHOW_DEPENDENCY_INSTALL)) {
-      hasInstalledDependencies = await promptInstallPythonDependencies(context, pythonExecutable);
-      if (hasInstalledDependencies) {
-        await vscode.workspace
-          .getConfiguration()
-          .update(CONFIG.SHOW_DEPENDENCY_INSTALL, false);
-      }
+      // check if ./out/python_libs exists; if not, the dependencies 
+      // for adafruit_circuitpython are not (successfully) installed yet
+      hasInstalledDependencies = fs.existsSync(pathToLibs) || await promptInstallPythonDependencies(context, pythonExecutable, pathToLibs);
```

Comment:
This is part is a bit tricky. The current check for installed dependencies is probably going to work most of the time.
Do you know what would happen if we changed requirements (is this a likely scenario?) or if a user manually removed some of the dependencies in `./out/python_libs`?

Reply:
Ideally, the project would be re-compiled when requirements are changed. Currently, the re-compilation process doesn't delete ./out/python_libs, but I can just remove that line.

It is a sorta cumbersome process to compile every time, but I'm sure that updating requirements won't happen all the time? What do you think?

Reply:
Thanks for clarifying. It sounds less of an issue than I thought at first. I would assume requirements wouldn't change frequently, if at all.
Up to you!

Reply:
Maybe you can try testing this out? E.g. what happens if we delete just one of the dependencies from `./out/python_libs`. I think the worst case scenario for this is the user having a "corrupt" environment and not being able to diagnose why.

If I understood correctly, if we delete this line we'll always prompt the user to install dependencies -- is that correct?

Reply:
I believe that you're correct- the user would have a corrupt environment. In order to fix this, they can just re-compile.

Reply:
That should be all right for now! But it might be worth adding an issue so we can handle this better in the future? If people end up hitting this particular scenario they will be able to upvote from there.

Reply:
Yeah, that'd be a good idea! I'll create an issue for it.